src/structures/CommandInteraction.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+const APIMessage = require('./APIMessage');
+const Interaction = require('./Interaction');
+const Snowflake = require('../util/Snowflake');
+
+/**
+ * Represents a command interaction, see {@link InteractionClient}.
+ * @extends {Interaction}
+ */
+class CommandInteraction extends Interaction {
+  constructor(client, data, syncHandle) {
+    super(client, data);
+
+    this.syncHandle = syncHandle;
+
+    /**
+     * The ID of the invoked command.
+     * @type {Snowflake}
+     * @readonly
+     */
+    this.commandID = data.data.id;
+
+    /**
+     * The name of the invoked command.
+     * @type {string}
+     * @readonly
+     */
+    this.commandName = data.data.name;
+
+    /**
+     * The options passed to the command.
+     * @type {Object}
+     * @readonly
+     */
+    this.options = data.data.options;
+  }
+
+  /**
+   * The timestamp the interaction was created at.
+   * @type {number}
+   * @readonly
+   */
+  get createdTimestamp() {
+    return Snowflake.deconstruct(this.id).timestamp;
+  }
+
+  /**
+   * The time the interaction was created at.
+   * @type {Date}
+   * @readonly
+   */
+  get createdAt() {
+    return new Date(this.createdTimestamp);
+  }
+
+  /**
+   * Acknowledge this interaction without content.
+   * @param {Object} [options] Options
+   */
+  async acknowledge(options = {}) {
+    await this.syncHandle.acknowledge(options);
+  }
+
+  /**
+   * Reply to this interaction.
+   * @param {(StringResolvable | APIMessage)?} content The content for the message.
+   * @param {(MessageOptions | MessageAdditions)?} options The options to provide.
+   */
+  async reply(content, options) {
+    let apiMessage;
+
+    if (content instanceof APIMessage) {
+      apiMessage = content.resolveData();
+    } else {
+      apiMessage = APIMessage.create(this, content, options).resolveData();
+      if (Array.isArray(apiMessage.data.content)) {
+        throw new Error('Message is too long');
+      }
+    }
+
+    const resolved = await apiMessage.resolveFiles();
+
+    if (!this.syncHandle.reply(resolved)) {
+      const clientID = this.client.interactionClient.clientID || (await this.client.getApplication()).id;
```

Comment:
Shouldn't this be fetchApplication()?

Reply:
yes but after #5147 they'll be just `this.client.application.id` since Discord sends the application id on ready but djs doesn't cache it yet